### PR TITLE
boards: stk3200: add support

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description

This PR add supports for the Silicon Labs STK3200, which I already PR'd before in #8295. The files are similar to the SLTB001a and STK3600, STK3700 and SLSTK3401a already supported. This is a Cortex M0+ based board, with a low-power memory LCD that is supported by U8g2.

I can provide remote access to the hardware, if needed.

### Issues/PRs references

#8585
#8520
#8630
#8295

### CODE REVIEW

**Comment by basilfx** [comment]
Updated this PR per #8873.

**Comment by tcschmidt** [comment]
Ping @kYc0o - can you review this?

File: boards/stk3200/Makefile.dep [hunk 1]
```diff
@@ -0,0 +1,11 @@
+ifneq (,$(filter saul_default,$(USEMODULE)))
+  USEMODULE += saul_gpio
+endif
+
+# limit the stack size so most tests and examples compile
+CFLAGS += -DTHREAD_STACKSIZE_DEFAULT=512
```
Comment (kYc0o) [inline boards/stk3200/Makefile.dep]: with #9356 maybe we can handle this?

**Comment by dylad** [comment]
@basilfx Could you rebase to the current master please ?
This way I can try and merge it :)

**Comment by dylad** [comment]
ping @basilfx

**Pull request closed without merging**